clippy_config/src/conf.rs

```diff
@@ -633,6 +633,20 @@ define_Conf! {
     /// exported visibility, or whether they are marked as "pub".
     #[lints(pub_underscore_fields)]
     pub_underscore_fields_behavior: PubUnderscoreFieldsBehaviour = PubUnderscoreFieldsBehaviour::PubliclyExported,
+    /// Whether to include functions outside of `#[cfg(test)]` in the linting process or not.
+    ///
+    /// This option allows running the lint against the integration tests: test functions located
+    /// there are not inside a node marked with `#[cfg(test)]` annotation (although they are
+    /// still marked using `#[test]` annotation and thus can have redundant "test_" prefix).
+    #[lints(redundant_test_prefix)]
+    redundant_test_prefix_check_outside_cfg_test: bool = false,
+    /// What suffix to use to avoid function name collisions when `test_` prefix is removed.
+    ///
+    /// If set to `"_works"`, the lint will suggest renaming `test_foo` to `foo_works`.
+    /// Suffix is added only when there is a collision with an existing function name,
+    /// otherwise just `test_` prefix is removed (and no suffix added).
+    #[lints(redundant_test_prefix)]
+    redundant_test_prefix_custom_suffix: String = String::from("_works"),
```

Comment:
I don't think we should create machine applicable suggestions for these cases. Mostly because I don't see a good universal way. The default will look weird for tests that actually try to panic:

```rust
    #[test]
    #[should_panic(expected = "explicit panic")]
    fn test_invalid_indexing() {
        panic!("explicit panic");
    }
```

The suggested name `invalid_indexing_works()` doesn't make much sense. It feels like `_test` would be the most universal prefix, but removing the prefix to then have it as a suffix would be weird.

I'm guessing that collisions after the renaming are rare. And if they happen it should be fine to just emit a warning and expect the user to come up with a new name.

Reply:
Yep, I've had the same feeling that it is not the most elegant approach. So, ignoring in auto-fix (but still warning about it) makes total sense.